### DISCUSSION TITLE
Fixing angular mean different from 0

### DISCUSSION THREAD

**Discussion by remy3896** [bmcclintock/momentuHMM#141] [Q&A] [answered]
Hello,

I am working on GPS data collected on sheep. Looking at the distribution of turning angles, I can tell that, at least for some behavioural states, the distribution is centred on PI and not 0.
To account for this effect, I need to specify to `fitHMM` that the mean for one state is PI (in my case, the first state).
`fitHMM` seems to only accept user-specified means if `estAngleMean` is set to `TRUE` for the relevant data stream ("angle" in my case).
I thus had to set the angle means as fixed parameters for both states.
Here is my code:

`dist <- list(step = "gamma", angle = "vm")

Par0_2s <- list(step = c(10, 130, 20, 50), angle = c(pi, 0, 0.1, 3))
fixPar_2s <- list(angle = c(pi, 0, NA, NA))

hmm1 <- fitHMM(data_hmm1, nbStates = 2, dist = dist, Par0 = Par0_2s, estAngleMean=list(angle=TRUE), fixPar = fixPar_2s)`

The optimisation runs without issue.
However, the computed parameter angle means are slightly different from PI and 0 (2.9345068 for state 1 and 0.06353729 for state 2).
More surprising, when I run the same code without the `fixPar` parameter, I get different values (3.0594606 and 0.0522979).
Thus, it seems that the specified fixed parameters are taken into account, but not to the point that the parameters are actually fixed.

How can I keep the angular means for both states actually fixed during the optimisation and still have a mean different from 0 for some states?

I’m using momentuHMM version 1.5.4 in R version 4.1.3.

Thank you in advance for your time.

**Comment by bmcclintock** [comment, marked as answer]
It's not clear to me why you would want to fix the `mean` parameter to `pi` instead of estimating it (or just fixing both `mean` parameters to zero by setting `estAngleMean` to `FALSE`, thereby allowing the `concentration` parameters to be easily interpretable in terms of directional persistence). Nevertheless, when `estAngleMean` is `TRUE` and `DM` is not specified for a circular data stream distribution (e.g. `vm`, `wrpcauchy`), {momentuHMM} uses a trick (also used in {moveHMM}) to make the optimization of the parameters more stable by maximizing the likelihood with respect to `x` and `y` below:
```
mean <- Arg(x + (0+1i) * y)
concentration <- sqrt(x^2 + y^2)
```
where both `x` and `y` have a lower bound of `-Inf` and upper bound depending on whether the angle distribution is "vm" (`Inf`) or "wrpcauchy" (`1`). Initial values for `x` and `y` are internally calculated as:
```
x <- concentration * cos(mean)
y <- concentration * sin(mean)
````
Thus when fixing the `mean` parameter using the `fixPar` argument, you can actually only fix "part" of the transformed parameters (because they are a function of both the `mean` and the `concentration` parameters). 

A workaround would involve specifying the `DM` argument for the `angle` distribution (which overrides the transformation) and specify the parameters on the working scale:
```
DM <- list(angle=list(mean=~1, concentration=~1)

Par0_2s <- list(step = c(10, 130, 20, 50), angle = c(tan(pi/2), tan(0/2), log(0.1), log(3)))
fixPar_2s <- list(angle = c(tan(pi/2), tan(0/2), NA, NA))

hmm1 <- fitHMM(data_hmm1, nbStates = 2, dist = dist, DM=DM, Par0 = Par0_2s, estAngleMean=list(angle=TRUE), fixPar = fixPar_2s)`
```

{momentuHMM} probably should not let users specify `fixPar$angle` when `estAngleMean$angle=TRUE` and `DM$angle=NULL`. In future releases, I'll plan to add an error trap for when this is done. Thanks for bringing this to my attention.

**Reply by remy3896** [reply]
Thank you for your answer, it works exactly as expected!

I don’t want let the optimization estimate the mean parameter because I am looking for a parsimonious model (and there is no reason for it not to be symmetrical left/right).
And with both mean parameters fixed to zero the model would not be able to fit properly to the data as my turning angle distribution is clearly bimodal with a peak centred in zero and one centred on pi.
